### PR TITLE
[solver]: 11 solver for large linear systems

### PR DESCRIPTION
**Pull request by clambr**
Complete all tasks of this issue.

### CODE REVIEW

File: source/inverseproblemmodules/adjoint_method_solver.py [hunk 1]
```diff
@@ -56,4 +57,27 @@ def _get_lambda_vector(self, inversemodel, flownetwork):
         lambda_vector = spsolve(csc_matrix(flownetwork.system_matrix.transpose()), -inversemodel.d_f_d_pressure)
         return lambda_vector
 
-# Todo: Implement other solvers (cg, amg, ...), do when necessary
\ No newline at end of file
+class AdjointMethodSolverPyAMG(AdjointMethodSolver):
+    """
+    Class for updating the gradient in alpha space with a sparse direct solver.
```
Comment (epprob) [inline source/inverseproblemmodules/adjoint_method_solver.py:62]: Should be sparse amg solver in description

File: testcases/testcase_inverse_problem.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 PARAMETERS = MappingProxyType(
     {
         # Setup parameters for blood flow model
-        "read_network_option": 1,  # 1: generate hexagonal graph
+        "read_network_option": 2,  # 1: generate hexagonal graph
```
Comment (epprob) [inline testcases/testcase_inverse_problem.py:26]: Maybe always reset the standard testcases to a standard test case with a hexagonal network. for fast testing.

**Review by epprob — CHANGES_REQUESTED** [review]
Hi Chryso. I commented on a few points. Mainly minor changes. But gives us a good opportunity to test the entire pull-request reviewing procedure..

File: source/inverseproblemmodules/adjoint_method_solver.py [hunk 3]
```diff
@@ -56,4 +57,27 @@ def _get_lambda_vector(self, inversemodel, flownetwork):
...
+        :type inversemodel: source.inverse_model.InverseModel
+        :param flownetwork: flow network object
+        :type flownetwork: source.flow_network.FlowNetwork
+        :returns: Lambda vector
+        :rtype: 1d numpy array
+        """
+        ml = smoothed_aggregation_solver(csr_matrix(flownetwork.system_matrix))  # AMG solver
+        M = ml.aspreconditioner(cycle='V')  # preconditioner
+        if inversemodel._lambda is None:
+            inversemodel._lambda, _ = cg(flownetwork.system_matrix.transpose(), -inversemodel.d_f_d_pressure, tol=1e-10,
```
[the remark this answers is not in the document]
Comment (clambr) [reply source/inverseproblemmodules/adjoint_method_solver.py:78]: I have checked it again. In the case of the lambda vector, it is not necessary to have two different tolerance values. It is quite fast for low tolerance in both cases (with/without initial values). However, I noticed that if I increase the tolerance to 1E-08, the solution is more or less the same as the previous tolerance (1E-10) but a bit faster. 
So, I'll keep 1E-08 as a tolerance value.

File: source/inverseproblemmodules/adjoint_method_solver.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (clambr) [reply source/inverseproblemmodules/adjoint_method_solver.py:62]: Done!

File: testcases/testcase_blood_flow_model.py [hunk 4]
```diff
@@ -17,7 +17,7 @@
 PARAMETERS = MappingProxyType(
     {
         # Setup parameters for blood flow model
-        "read_network_option": 1,  # 1: generate hexagonal graph
+        "read_network_option": 2,  # 1: generate hexagonal graph
```
[the remark this answers is not in the document]
Comment (clambr) [reply testcases/testcase_blood_flow_model.py:20]: Done!

File: testcases/testcase_inverse_problem.py (hunk 2 above)
Comment (clambr) [reply testcases/testcase_inverse_problem.py:26]: Done!

File: testcases/testcase_inverse_problem.py [hunk 5]
```diff
@@ -68,15 +70,17 @@
...
+                                    # 3-...: other solvers
         # Target edges
         "csv_path_edge_target_data": "data/inverse_model/edge_target.csv",
         # Parameter edges
         "csv_path_edge_parameterspace": "data/inverse_model/edge_parameters.csv",
         # Gradient descent options:
         "gamma": .5,
         "phi": .5,
-        "max_nr_of_iterations": 50
+        "max_nr_of_iterations": 5
```
[the remark this answers is not in the document]
Comment (clambr) [reply testcases/testcase_inverse_problem.py:83]: Done!

File: data/inverse_model/edge_target.csv [hunk 6]
```diff
@@ -3,4 +3,5 @@ edge_tar_eid,edge_tar_type,edge_tar_value,edge_tar_range_pm,edge_tar_sigma
 19,1,4E-18,4E-18,1.00E-18
 20,1,1E-17,0,1.00E-18
 25,2,3E-07,0,1.00E-07
-29,2,-1E-07,1E-07,1.00E-07
```
[the remark this answers is not in the document]
Comment (clambr) [reply data/inverse_model/edge_target.csv:6]: Done!

**Review by epprob — APPROVED** [review]

**Pull request merged**